l1-contracts/src/core/messagebridge/MessageBox.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {IRegistryReader} from "./IRegistryReader.sol";
+
+/**
+ * @title MessageBox
+ * @author Aztec Labs
+ * @notice Data structure used in both Inbox and Outbox for keeping track of entries
+ * Implements a multi-set storing the multiplicity (count for easy reading) at the entry.
+ */
+abstract contract MessageBox {
+  error MessageBox__Unauthorized();
+  error MessageBox__NothingToConsume(bytes32 entryKey);
+  error MessageBox__OversizedContent();
+
+  /// @dev Actor on L1. ChainID if multiple L1s tap into the same Aztec instance.
+  struct L1Actor {
+    address actor;
+    uint256 chainId;
+  }
+
+  /// @dev Actor on L2. `version` specifies which Aztec instance the actor is on (useful for upgrades)
+  struct L2Actor {
+    bytes32 actor;
+    uint256 version;
+  }
+
+  /**
+   * @dev Entry struct - Done as struct to easily support extensions if needed
+   * @param count - The occurrence of the entry in the dataset
+   * @param fee - The fee provided to sequencer for including in the inbox. 0 if Oubox (as not applicable).
+   */
+  struct Entry {
+    uint64 count;
+    uint64 fee;
+    uint32 deadline;
+  }
+
+  IRegistryReader registry;
```

Comment:
Missing visibility. Should also be `immutable` as cannot be updated beyond constructor. Remember that `constant` and `immutable` variables should be `CAPITAL_CASE`. 

Ordering of variables should be updated. Constants -> immutables -> mutable data.